docs/quality-gates/IntegrationWithCICD/IntegrationWithAzureDevOps.md

```diff
@@ -0,0 +1,154 @@
+---
+sidebar_label: Integration with Azure DevOps
+---
+
+# Integration with Azure DevOps
+
+This tutorial guides you through the process of integrating ReportPortal with Azure DevOps for Node.js projects. Depending on your project, you can use either a Bash (Linux) or Powershell script. This pipeline includes steps for the application build, running tests, capturing the ReportPortal Launch ID, and implementing a Quality Gate check.
+
+## Prerequisites
+
+Before you begin, ensure you have:
+
+- An Azure DevOps account set up.
+- ReportPortal account and a project created.
+
+## Azure DevOps Pipeline Overview
+
+The Azure DevOps pipeline for integrating ReportPortal consists of the following key steps:
+
+1. **Run application build and tests:** Execute npm tests.
+2. **Capture Launch ID:** Extract the ReportPortal Launch ID from the test execution logs.
+3. **Quality Gate:** Check the Quality Gate status on ReportPortal using the captured Launch ID.
+
+## Configuring Azure DevOps pipeline
+
+### Environment Variables
+
+Set the following environment variables in your Azure DevOps Pipeline:
+
+- `RP_INSTANCE_URL`: The URL of your ReportPortal instance.
+- `RP_PROJECT`: The ReportPortal project name.
+- `REQUEST_TIMEOUT_SECONDS`: Timeout for ReportPortal API requests.
+- `SCRIPT_TIMEOUT_SECONDS`: Timeout for the entire script execution.
+
+In the **Variables** section, add the following sensitive variable:
+- `RP_API_KEY`: Your API key from ReportPortal.
+
+### Step 1: Run application build and tests
+
+Use the following scripts to run tests with npm. The test execution logs are captured in `./console.log`. Then the ReportPortal Launch ID is extracted and saved in `RP_LAUNCH_ID`. The script you choose depends on whether you are using Bash or Powershell. 
+
+#### Bash Script
+
+```yaml
+- script: |
```

Comment:
The step should not fail if we talk about Quality Gates configuration. You put try-catch in powershell script but left bash script without such option.

Reply:
Fixed please check